File: ts/896/852.ts
function peakIndexInMountainArray(arr: number[]): number {
  let l = 1, r = arr.length - 2, ans = 0
  while (l <= r) {
    let m = l + ((r - l) >> 1)
    if (arr[m - 1] < arr[m]) {
      l = m + 1
      ans = m
    } else {
      r = m - 1
      ans = m - 1
    }
  }
  return ans
};

File: py3/896/852.py
from typing import List

class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        l, r = 1, len(arr) - 2
        while l <= r:
            m = l + (r - l) // 2
            if arr[m] < arr[m + 1]:
                l = m + 1
            else:
                r = m - 1
        return l